pkg/distributor/distributor.go

```diff
@@ -1013,11 +1024,13 @@ func (d *Distributor) limitsMiddleware(next push.Func) push.Func {
 
 		il := d.getInstanceLimits()
 		if il.MaxInflightPushRequests > 0 && inflight > int64(il.MaxInflightPushRequests) {
+			d.discardedInstanceRequests.WithLabelValues(validation.ReasonDistributorMaxInflightPushRequests).Inc()
```

Comment:
Similar to my previous comment on "discard", these are not "validation" errors. Validation errors indicate problem in the request / tenant limit.

Instead of introducing new "reasons" in `validation` package, I'd suggest to use something like

```suggestion
			d.discardedInstanceRequests.WithLabelValues(globalerror.DistributorMaxInflightPushRequests.String()).Inc()
```

(This requires new `String()`  method)

Reply:
I can move it but I put them in `validation` because that's where the label-form for other `globalerror` errors live. I wanted to keep all the labels for error metrics in the same place.

Reply:
Added a `.LabelValue()` to `globalerror.ID` which is now used everywhere.